src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -59,18 +69,27 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo taskInfo) {
           .setTaskId(journalNodeTask.taskInfo.getTaskId())
           .setState(TaskState.TASK_RUNNING)
           .build());
+      healthCheckJN = new TimedHealthCheck(driver, journalNodeTask);
+      timer.scheduleAtFixedRate(healthCheckJN, schedulerConf.getHealthCheckWaitingPeriod(),
+          schedulerConf.getHealthCheckFrequency());
```

Comment:
After further discussion, it looks like we can do all health checks through http for everything except zkfc. We can use the Mesos API for this.